cloud/user/user_test.go

```diff
@@ -28,88 +28,89 @@ func (t testWriter) Write(p []byte) (n int, err error) {
 
 func TestCreateInvite(t *testing.T) {
 	testUtil.InitTestConfig(testUtil.CloudPlatform)
-	expectedInput := astro.CreateUserInviteInput{
-		InviteeEmail:   "test-email@test.com",
-		Role:           "ORGANIZATION_MEMBER",
-		OrganizationID: "test-org-id",
-	}
-	expectedInvite := astro.UserInvite{
-		UserID:         "test-user-id",
-		OrganizationID: "test-org-id",
-		OauthInviteID:  "test-oauth-invite-id",
-		ExpiresAt:      "test-expiry",
+	inviteUserID := "user_cuid"
+	createInviteResponseOK := astrocore.CreateUserInviteResponse{
+		HTTPResponse: &http.Response{
+			StatusCode: 200,
+		},
+		JSON200: &astrocore.Invite{
+			InviteId: "",
+			UserId:   &inviteUserID,
+		},
 	}
-	expectedWorkspace := astro.Workspace{
-		ID:             "test-workspace-id",
-		Label:          "test-workspace-label",
-		OrganizationID: "test-org-id",
+	errorBody, _ := json.Marshal(astrocore.Error{
+		Message: "failed to create invite: test-inv-error",
+	})
+	createInviteResponseError := astrocore.CreateUserInviteResponse{
+		HTTPResponse: &http.Response{
+			StatusCode: 500,
+		},
+		Body:    errorBody,
+		JSON200: nil,
 	}
 	t.Run("happy path", func(t *testing.T) {
 		expectedOutMessage := "invite for test-email@test.com with role ORGANIZATION_MEMBER created\n"
 		out := new(bytes.Buffer)
-		mockClient := new(astro_mocks.Client)
-		mockClient.On("GetWorkspace", mock.Anything).Return(expectedWorkspace, nil).Once()
-		mockClient.On("CreateUserInvite", expectedInput).Return(expectedInvite, nil).Once()
+		mockClient := new(astrocore_mocks.ClientWithResponsesInterface)
+		mockClient.On("CreateUserInviteWithResponse", mock.Anything, mock.Anything, mock.Anything).Return(&createInviteResponseOK, nil).Once()
 		err := CreateInvite("test-email@test.com", "ORGANIZATION_MEMBER", out, mockClient)
 		assert.NoError(t, err)
 		assert.Equal(t, expectedOutMessage, out.String())
 	})
-	t.Run("error path when CreateUserInvite returns an error", func(t *testing.T) {
+	t.Run("error path when CreateUserInviteWithResponse returns an error", func(t *testing.T) {
 		expectedOutMessage := "failed to create invite: test-inv-error"
 		out := new(bytes.Buffer)
-		mockClient := new(astro_mocks.Client)
-		mockClient.On("GetWorkspace", mock.Anything).Return(expectedWorkspace, nil).Once()
-		mockClient.On("CreateUserInvite", expectedInput).Return(astro.UserInvite{}, errorInvite).Once()
+		mockClient := new(astrocore_mocks.ClientWithResponsesInterface)
+		mockClient.On("CreateUserInviteWithResponse", mock.Anything, mock.Anything, mock.Anything).Return(&createInviteResponseError, nil).Once()
```

Comment:
We loose the ability to test that our input to the API call was exactly what we expected. For e.g. if we want/need to test that email is always lower case, etc we can do that easily if we didn't have a `mock.Anything` but had the `expectedInput` instead.

Reply:
yep, we pass it directly to api so i thought there's not too much value of testing this ( or if we did normalize the email it can be tested purposely ). I am fine to switch back to use input compare here though